src/CFNetwork/CFHTTPMessage.cs

```diff
@@ -80,16 +80,20 @@ static IntPtr GetVersion (Version version)
 				return _HTTPVersion1_1.Handle;
 			else if (version.Equals (HttpVersion.Version10))
 				return _HTTPVersion1_0.Handle;
-			else if (version.Major == 2 && version.Minor == 0) {
+			else if (version.Major == 3 && version.Minor == 0) {
+				if (_HTTPVersion3_0 != IntPtr.Zero)
+					// HTTP 3.0 requires OS X 10.16 or later.
+					return _HTTPVersion3_0;
+				else if (_HTTPVersion2_0 != null && _HTTPVersion2_0.Handle != IntPtr.Zero)
+					// HTTP 2.0 requires OS X 10.11 or later.
+					return _HTTPVersion2_0.Handle; 
+				else
+					return _HTTPVersion1_1.Handle;
+			} else if (version.Major == 2 && version.Minor == 0) {
```

Comment:
better :) but I think it could be simplified

First `.Handle` is not needed for the other, already existing, `NSString` fields. They are internal so this would not be a breaking change.

Next, you can reduce the number of `if` by removing `else`

```
if (== 3.0) {
	// HTTP 3.0 requires OS X 10.16 or later.
    if (_HTTPVersion3_0 != IntPtr.Zero)
 		return _HTTPVersion3_0;
}
if (>= 2.0) {
    // this will catch 3.0 on system where it's not supported and return 2.0
    if (_HTTPVersion2_0 != IntPtr.Zero)
 		return _HTTPVersion2_0;
}
if (>= 1.1) {
   // ...
}
...
```

Reply:
Did not change the other ones because the NSStrings are indeed used and did want to minimize the diff. Do you want me to change all occurrences to use handles and create the native string when needed instead?

Reply:
are they used _many_ time elsewhere ? a quick grep suggest noe
if it's only in this method then it makes sense, otherwise it depends on how large the scope become

Reply:
is not too bad, one small change in another method from one of the other internal vars.